streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -979,6 +1004,11 @@ boolean maybeCommit() {
         } else {
             committed = taskManager.maybeCommitActiveTasksPerUserRequested();
             if (committed > 0) {
+                if (eosBetaEnabled) {
```

Comment:
For line 1005 above: this is not introduced in this PR, but I'm wondering if we should check for `maybeCommitActiveTasksPerUserRequested` still in `maybeCommit`, since we've already call it after each iteration and users can only request committing either during `process` or `punctuate`, so it seems the second `maybeCommit` do not need to check for this any more?

Just leaving it here but we do not need to piggy-back it into this PR -- if you agrees we can just leave a TODO marker.

Reply:
I agree that we can do some cleanup here -- but I think atm it's correct, because in `runOnce()` we check for user requested commits only if we did `process()` data, but not if we did `punctuate()` -- hence, this check currently seems to ensure that we commit on user request issued within `punctuate`?

Actually, I am also not sure why we call `maybeCommit()` at the end of `runOnce()`?